src/baseShape.ts

```diff
@@ -82,14 +84,48 @@ export class InteractiveObject {
 
   public mouseClick: Vertex = null;
 
-  public isHovered: boolean = false;
-  public isClicked: boolean = false;
+  private _isHovered: boolean = false;
+  public get isHovered(): boolean{
+    return this._isHovered;
+  };
+  set isHovered(hovered: boolean) {
+    if (hovered !== this._isHovered && this.referencePath !== "#") {
+      // The first check is important, otherwise we fire the event every mouse move.
+      // The second is for dev purpose, should we keep it ?
+      const highlightData = {
+        referencePath: this.referencePath,
+        highlight: hovered,
+        select: false
+      }
+      highlightShape.next(highlightData);
+    }
+    this._isHovered = hovered;
+  }
+
+  private _isClicked: boolean = false;
+  public get isClicked(): boolean {
+    return this._isClicked;
+  }
+  set isClicked(clicked: boolean) {
+    if (clicked != this._isClicked && this.referencePath !== "#") {
+      // Same than isHovered
+      const highlightData = {
+        referencePath: this.referencePath,
+        highlight: clicked,
+        select: true
+      }
+      highlightShape.next(highlightData);
+    }
+    this._isClicked = clicked;
+  }
```

Comment:
What do you mean by complicated ? This seems pretty minimal to me.

By convention, I always place my attributes (and therefore getters/setters, because they still are attributes) before the constructor, but I actually don't mind to change them as it is just personnal preference without any actual further impact on the code. Here is my covention, probably due to angular bias : 

```ts
class MyClass {
    // Attributes
    private attribute1: any = null;

    // Getters/Setters
    private _attribute2: any = null;
    public get attribute2(): any {
        return this._attribute2;
    }
    public set attribute2(value: any) {
        this._attribute2 = value;
    }

    constructor() { }

    // Methods
    function myMethod() {
        // Do Stuff
    }
}
```

I don't know if there is any globally accepted convention about this, so let me know if I should still change this after the given elements.

Reply:
I find it complicated because i don't understand what you are doing with highlightShape. 
I understand it is a global constant which attributes are changed with mouse state attributes at any change in a shape but i don't understand when it is used and how.

Reply:
And i prefer to have getters and setters after the constructor in order to have a global simple overview on class attributes. 

Like you, i'm less an expert than you (here i just created an infinite loop) so do what you think is better, even if it is my baby.